parsel/selector.py

```diff
@@ -722,25 +729,30 @@ def get(self) -> Any:
         For HTML and XML, the result is always a string, and percent-encoded
         content is unquoted.
         """
-        if self.type in ("text", "json"):
+        if self.type in ("json", "text"):
+            if self.type == "text" and self._text_lazy_html_root is not None:
+                return etree.tostring(
+                    self._text_lazy_html_root, encoding="unicode", with_tail=False
+                )
```

Comment:
A problem with this approach is that we are assuming HTML, when it could be XML.